R/get_p_value.R

```diff
@@ -198,7 +199,20 @@ two_sided_p_value <- function(vec, obs_stat) {
   min(raw_res, 1)
 }
 
-
+check_x_vs_obs_stat <- function(x, obs_stat) {
+  # check if x and obs_stat might have been mistakenly supplied
+  # in the reverse order
+  if (is_generated(obs_stat) &&
+      !is_generated(x)) {
+    stop_glue(
+      "The `obs_stat` argument has been passed to `generate()` while the ",
+      "`x` argument has not; this is an untested workflow. ", 
+      "Have you mistakenly switched those arguments?"
```

Comment:
I think this should emphasize that `obs_stat` was passed as the "first" argument, and that's not what we want. How about the following?

> The `obs_stat` argument has been passed to `generate()` as the first argument when `generate()` expects `x` as the first argument instead. Have you mistakenly switched the order of `obs_stat` and `x`?

My concern is will the user know what `x` is from this sentence? I realize `x` is the first argument of `generate()` in the documentation so there shouldn't be a mystery once someone looks at the documentation, but is the message itself clear enough?

Reply:
I like the form of this message! Confirming that those `generate()`s should be `get_p_value()`?

Reply:
yes!